interfaces/builtin/opengl.go

```diff
@@ -191,14 +206,48 @@ var openglConnectedPlugUDev = []string{
 	`KERNEL=="galcore"`,
 }
 
+func (iface *openGlInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	spec.AddSnippet(openglConnectedPlugAppArmor)
+
+	// Allow mounting the Nvidia driver profiles directory
+	if !osutil.IsDirectory(dirs.NvidiaProfilesDir) {
+		return nil
+	}
+
+	spec.AddUpdateNSf(`  # Read-only access to Nvidia driver profiles in %[2]s
+	mount options=(bind) /var/lib/snapd/hostfs%[1]s/ -> %[2]s/,
+	remount options=(bind, ro) %[2]s/,
+	umount %[2]s/,
+  
+  `, dirs.NvidiaProfilesDir, dirs.StripRootDir(dirs.NvidiaProfilesDir))
+
```

Comment:
You'll probably also need this code:

```suggestion
	apparmor.GenWritableProfile(spec.AddUpdateNSf, dirs.StripRootDir(dirs.NvidiaProfilesDir), 3)
```

This will generate the AppArmor rules necessary for snap-update-ns to make a `/usr/share/nvidia` directory in the sandbox mount namespace for the bind mount to attach to.

Reply:
I tested it with a snap with only a passthrough script and an `opengl` plug, and the directory exists as expected. I can add this, but it doesn't *seem* to be necessary unless that only applies under certain conditions.

Reply:
im adding this because the spread tests failed on a few required systems, it seems like only certain ones may require it?

Reply:
hmm, still failing not sure why

Reply:
Did your test snap have any layouts to paths under `/usr/share`? That could have generated the same AppArmor rules this `GenWritableProfile` call would.

The test snaps used by the interfaces-opengl-nvidia spread test don't have use layouts, so wouldn't mask that kind of problem.